### PR TITLE
added mongodb services with minimal changes existed flow

### PR DESCRIPTION
**Pull request by DFSko**

### CODE REVIEW

**Review by mingazhev — APPROVED** [review]

File: Api/Infrastructure/MongoDb/Extensions/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+{
+    public static class ServiceCollectionExtensions
+    {
+        public static IServiceCollection AddMongoDbStorage(this IServiceCollection services, IConfiguration configuration)
+        {
+            BsonSerializer.RegisterSerializationProvider(new SerializationProvider());
+
+            services.Configure<MongoDbOptions>(o =>
+            {
+                o.ConnectionString = configuration.GetValue<string>(configuration.GetValue<string>("MongoDB:ConnectionString"));
```
Comment (kirillta) [inline Api/Infrastructure/MongoDb/Extensions/ServiceCollectionExtensions.cs:20]: That wouldn't such simple: a base have a password

File: Api/Infrastructure/MongoDb/Services/AvailabilityStorage.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        private static async Task<IMongoCollection<AccommodationAvailabilityResult>> GetOrCreateCollection(IMongoDbClient client)
+        {
+            var collection = client.GetDatabase().GetCollection<AccommodationAvailabilityResult>(nameof(AccommodationAvailabilityResult));
+
+            var searchIndexDefinition = Builders<AccommodationAvailabilityResult>.IndexKeys.Combine(
+                Builders<AccommodationAvailabilityResult>.IndexKeys.Ascending(f => f.SearchId),
+                Builders<AccommodationAvailabilityResult>.IndexKeys.Ascending(f => f.Supplier));
+
+            var ttlIndexDefinition = Builders<AccommodationAvailabilityResult>.IndexKeys.Ascending(f => f.Created);
+            var ttlIndexOptions = new CreateIndexOptions {ExpireAfter = TimeSpan.FromMinutes(45)};
```
Comment (kirillta) [inline Api/Infrastructure/MongoDb/Services/AvailabilityStorage.cs:40]: Probably, it's better move the number to a self-explaining constant

File: Api/Infrastructure/MongoDb/Extensions/ServiceCollectionExtensions.cs (hunk 1 above)
Comment (DFSko) [reply Api/Infrastructure/MongoDb/Extensions/ServiceCollectionExtensions.cs:20]: Connection string may contain password

mongodb://[username:password@]host1[:port1][,...hostN[:portN]][/[defaultauthdb][?options]]

File: Api/Infrastructure/MongoDb/Services/AvailabilityStorage.cs (hunk 2 above)
Comment (DFSko) [reply Api/Infrastructure/MongoDb/Services/AvailabilityStorage.cs]: Fixed

**Pull request merged**